src/libtiled/mapobject.cpp

```diff
@@ -209,16 +209,40 @@ void MapObject::flip(FlipDirection direction)
             mCell.setFlippedVertically(!mCell.flippedVertically());
     }
 
-    if (!mPolygon.isEmpty()) {
-        const QPointF center2 = mPolygon.boundingRect().center() * 2;
+    //creating flipmatrix
+    QMatrix flipMatrix;
+    flipMatrix.translate(flipCenter.x(), flipCenter.y());
+    if (direction == FlipHorizontally)
+        flipMatrix.scale(1, -1);
```

Comment:
While possibly ambiguous, this changes the interpretation of what a horizontal flip is compared to what it was before. A horizontal flip is supposed to flip the left/right sides, whereas a vertical flip is supposed to flip top/bottom. This means horizontal flip should use a -1 scale on x rather than y.